src/Rotalia/UserBundle/Model/MemberCredit.php

```diff
@@ -36,8 +36,14 @@ public function adjustCredit($amount, $creditLimit = null)
         $amount = doubleval($amount);
         $newCredit = round($currentCredit + $amount, 2);
 
-        if ($creditLimit !== null && $newCredit < $creditLimit) {
-            throw new OutOfCreditException($newCredit, $creditLimit);
+        if ($creditLimit !== null) {
+            // The MemberCredit is per convent. Compare total credit against the credit limit
+            $totalCredit = doubleval($this->getMember()->getTotalCredit());
+            $newTotalCredit = round($totalCredit + $amount, 2);
+
+            if ($newTotalCredit < $creditLimit) {
+                throw new OutOfCreditException($newTotalCredit, $creditLimit);
```

Comment:
See case on muidugi varem ka katmata jäänud, aga oleks tore kui see oleks ka testitud. Seda võin ise teha hiljem.

Ma vaatan, et unit teste üldse meil pole, eks peab ette võtma ja neid ka tekitama.
Mul siin masteri korda ajamist need testid aitasid küll, et mingid apsakad kinni püüda. Samas 100% kindlust ikka pole praegu.

Reply:
Aga muidu paistab kõik hästi olevat, väga hea :)

Reply:
Lisasin `testPurchaseFailIfNotEnoughCredit`